build/ci-build.yml

```diff
@@ -85,26 +99,27 @@ stages:
               command: 'custom'
               custom: 'new'
               arguments: '-i "templates/*.nupkg"'
-          - task: DotNetCoreCLI@2
-            displayName: 'Create new project from template'
-            inputs:
-              command: 'custom'
-              custom: 'new'
-              arguments: 'arcus-webapi -n MyProject -o /MyProject'
-          - task: Docker@1
-            displayName: 'Build Docker image from newly created web API project from template'
-            inputs:
-              dockerFile: /MyProject/Dockerfile
-              imageName: 'myproject:$(Build.BuildId)'
-          - task: Docker@1
-            displayName: 'Run new web API project Docker image from template'
-            inputs:
-              command: 'Run an image'
-              imageName: 'myproject:$(Build.BuildId)'
-              containerName: 'myproject'
+          - template: 'templates/run-new-project-from-template.yml'
+            parameters:
+              projectName: 'Arcus.Demo.WebApi'
+              projectAlias: 'arcus-webapi'
               ports: '$(Http.Port):$(Http.Port)'
               envVars: |
                 ARCUS_HTTP_PORT=$(Http.Port)
+          - template: 'templates/run-new-project-from-template.yml'
+            parameters:
+              projectName: 'Arcus.Demo.ServiceBus.Queue'
+              projectAlias: 'arcus-servicebus-queue'
+              ports: '$(Arcus_Worker_HealthPort):$(Arcus_Worker_HealthPort)'
+              script: |
+                Get-Content '/Arcus.Demo.ServiceBus.Queue/Program.cs' | 
```

Comment:
Would you mind elaborating on this please?

Reply:
😄 also an interesting part.
A problem rises that doesn't existed in the web API template. The inital project that will be created from this template, will contain build errors; since we require a `ISecretProvider`.  We have two test suites of integration tests: Docker and Self-Containing.

The self-containing will solve this problem by changing the file on disk via code.
But the docker tests can't do this by code since we're in a YAML context. That's why we have to change some created files before we can correctly run the project. In this case the `Program.cs` where a `#error` line exists.

Reply:
How is this different from the WebAPI exactly because also over there we add the cached secret provider, no?

Reply:
Yeah, correct; but only if you selected a project option (like `--authentication SharedAccessKey`). Without any options, we would have a completely runnable project.

Reply:
🆗